### PR TITLE
Add Different Langs

### PR DESCRIPTION
**Pull request by CyberZenDev**
1 - Bash
2 - Rust
3 - Ruby
4 - TypeScript
5 - PhP
6 - VBS
7 - Dart

### CODE REVIEW

**Comment by ItsTato** [comment]
Howdy!
It seems none of these have comments yet, so you should add some
Other than that they seem pretty good!

**Comment by CyberZenDev** [comment]
Ok Sure How would you like me to comment

File: Lua/Plugins/rust.lua [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+highlight(":", "special")
+highlight(",", "special")
+
+--- Strings and Characters
+highlight_region('"', '"', "string")
+highlight_region("'", "'", "character")
+
+--- Comments
+highlight_region("//", "", "comment", true)
+highlight_region("/*", "*/", "comment", true)
```
Comment (RealPacket) [inline Lua/Plugins/rust.lua:97]:
```suggestion
highlight_region("/*", "*/", "comment", false)
```
the 3rd parameter for `highlight_region` determines if the highlight isn't multi-line.

File: Lua/Plugins/rust.lua [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+highlight(";", "special")
+highlight("_", "special")
+highlight("'", "special")
+highlight(":", "special")
+highlight(",", "special")
+
+--- Strings and Characters
+highlight_region('"', '"', "string")
+highlight_region("'", "'", "character")
+
```
Comment (RealPacket) [inline Lua/Plugins/rust.lua:94]:
```suggestion
--- IDE comments (CTRL + L)
add_comment("Your code is so garbage that it rusted away.")
add_comment("Every source code file you touch transforms into Rust code.")
add_comment('Translate "Rewrite it in Rust" from Ferris to English: "Screw your codebase, rewrite it all in my preferred language."')
-- AI
add_comment("Oh, you write Rust code? Well, I must admit, your coding skills are as sturdy and reliable as a rusty old bicycle. But hey, at least your code matches your sense of humor—dry and lacking in any real punchlines.")
```
You don't have any comments, I've suggested some comments.

**Review by RealPacket — CHANGES_REQUESTED** [review]
Fix a bug, add some comments, and done.

File: Lua/Plugins/rust.lua (hunk 1 above)
Comment (ItsTato) [reply Lua/Plugins/rust.lua:97]: 4th parameter
3rd is "comment" or "string"

=3

**Comment by uynilo9** [comment]
The file names are supposed to be the same as their file extensions. e.g. typescript -> ts, rust -> rs

File: Lua/Plugins/typescript.lua [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+highlight("=>", "special")
+
+--- Strings
+highlight_region('"', '"', "string")
+highlight_region("'", "'", "string")
+highlight_region("`", "`", "string")
+
+--- Comments
+highlight_region("//", "", "comment", true)
+highlight_region("/*", "*/", "comment", true)
```
Comment (uynilo9) [inline Lua/Plugins/typescript.lua:125]: The fourth arg (line_only) of `/* */` should be false

Comment (uynilo9) [inline Lua/Plugins/typescript.lua:125]:
```suggestion
highlight_region("/*", "*/", "comment", false)
```

File: Lua/Plugins/typescript.lua [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+highlight(",", "special")
+highlight("<", "special")
+highlight(">", "special")
+highlight(":", "special")
+highlight("?", "special")
+highlight("`", "special")
+highlight("=>", "special")
+
+--- Strings
+highlight_region('"', '"', "string")
```
Comment (uynilo9) [inline Lua/Plugins/typescript.lua:119]:
```suggestion
highlight_region('"', '"', "string", true)
```
I suppose this is optional

File: Lua/Plugins/typescript.lua [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+highlight_region("'", "'", "string")
```
Comment (uynilo9) [inline Lua/Plugins/typescript.lua:120]:
```suggestion
highlight_region("'", "'", "string", true)
```
This is as well

File: Lua/Plugins/rust.lua (hunk 1 above)
Comment (RealPacket) [reply Lua/Plugins/rust.lua:97]: > 4th parameter 3rd is "comment" or "string", etc
> 
> =3

oops

**Comment by Symmettry** [comment]
Typescript is missing editor comments, wrong file name, and is missing default variables & functions.

It's also worth noting that the type identifier ": " is not highlighted, and highlighting primitive values as reserved does not look very good.

Also, for all langs here, you don't need to highlight operators. the editor does it by default. Same with special characters. Unsure about < or > tho.

Typescript should probably be in similar flow to the javascript one as well so if anything is added it can be added to both just as easily.

**Review by QuinceTart10 — CHANGES_REQUESTED** [review]
1. bash uses the .sh extension, so the file should be renamed to `sh.lua`
2. there are no comments

**Review by QuinceTart10 — CHANGES_REQUESTED** [review]
Rust's file extension isn't `rust`, it's `rs`, so this file needs to be renamed to `rs.lua`

**Comment by Symmettry** [comment]
The highlight is "comments" not "comment"
Issue is in multiple files.

**Pull request closed without merging**